src/main/java/org/cbio/gdcpipeline/JobConfiguration/BatchConfiguration.java

```diff
@@ -72,19 +83,67 @@ public Tasklet setUpPipelineTasklet() {
 
     @Bean
     @StepScope
-    public Tasklet createFileMappings(){
-        return new FileMappingTasklet();
+    public Tasklet xmlFileMappingTasklet() {
+        return new XmlFileMappingTasklet();
+    }
+
+    @Bean
+    @JobScope
+    public JobExecutionDecider clinicalFileTypeDecider() {
+        return new ClinicalFileTypeDecider();
+    }
+
+    @Bean
+    public Flow configurePipelineFlow() {
+        return new FlowBuilder<Flow>("configurePipelineFlow")
+                .start(setUpPipeline())
+                .on("COMPLETED").to(clinicalFileTypeDecider())
+                .from(clinicalFileTypeDecider()).on("XML").to(xmlFileMappingStep())
+                .from(clinicalFileTypeDecider()).on("XLSX").fail()
+                .from(clinicalFileTypeDecider()).on("FAIL").fail()
+                .build();
+
+    }
+    @Bean
+    public Flow clinicalDataFlow() {
+        return new FlowBuilder<Flow>("clinicalDataFlow")
+                .start(clinicalDataStep)
+                .next(clinicalMetaDataStep)
+                .build();
+
     }
 
     @Bean
-    public Step runClinicalDataStep() {
-        return runClinicalDataStep;
+    public Flow mutationDataFlow() {
+        return new FlowBuilder<Flow>("mutationDataFlow")
+                .start(mutationDataStep)
+                .from(mutationDataStep).on("CONTINUE").to(mutationDataStep)
+                .next(mutationMetaDataStep)
+                .build();
+
+    }
+
+    @StepScope
+    public JobExecutionDecider stepDecider() {
+        return new StepDecider();
     }
 
+    @Bean
+    public Flow completeFlow() {
+        return new FlowBuilder<Flow>("completeFlow")
+                .start(clinicalDataFlow())
+                .next(mutationDataFlow())
+                .build();
+    }
 
     @Bean
-    public Step runClinicalMetaDataStep() {
-        return runClinicalMetaDataStep;
+    public Flow buildFlow() {
+        return new FlowBuilder<Flow>("buildFlow")
+                .start(stepDecider())
+                .on(StepDecider.STEP.ALL.toString()).to(completeFlow())
+                .on(StepDecider.STEP.CLINICAL.toString()).to(clinicalDataFlow())
+                .on(StepDecider.STEP.MUTATION.toString()).to(mutationDataFlow())
+                .build();
     }
```

Comment:
Currently I wrote it this way as I couldn't figure out how to handle a case when user inputs a subset of steps to run. I am thinking of something like a custom flow builder that will dynamically add flows depending on the input